dask/async.py

```diff
@@ -426,69 +426,69 @@ def get_async(apply_async, num_workers, dsk, result, cache=None,
         result_flat = set([result])
     results = set(result_flat)
 
-    dsk = dsk.copy()
-    for f in start_cbs:
-        f(dsk)
-
-    dsk, dependencies = cull(dsk, list(results))
+    try:
+        dsk = dsk.copy()
+        for f in start_cbs:
+            f(dsk)
```

Comment:
This opens the (small) chance that a callback will have `finish` called without having `start` ever called. The existing callbacks may need to be fixed to be robust against this case.